commit 8403fb28b1bc259c7d816e67602f1384634fab33
Author: Devlin Cronin <rdevlin.cronin@chromium.org>
Date:   Wed May 27 16:29:02 2026 -0700

    [Extensions] Remove old MV2 phase handling in developerPrivate
    
    Remove now-unnecessary handling in the developerPrivate API for pieces
    of the MV2 deprecation. Since this has been fully launched for >1 year,
    we no longer need the old UI implementations or individual extension
    acknowledgement.
    
    This CL should have no behavior change in practice, since the removed
    UIs are not actively shown to any users today.
    
    Bug: 431097630
    Change-Id: I2757555eae0be80df284abc372bd8e00e365b3a3
    Reviewed-on: https://chromium-review.googlesource.com/c/chromium/src/+/7876097
    Reviewed-by: Kelvin Jiang <kelvinjiang@chromium.org>
    Reviewed-by: Demetrios Papadopoulos <dpapad@chromium.org>
    Auto-Submit: Devlin Cronin <rdevlin.cronin@chromium.org>
    Commit-Queue: Devlin Cronin <rdevlin.cronin@chromium.org>
    Cr-Commit-Position: refs/heads/main@{#1637322}

diff --git a/chrome/browser/extensions/api/developer_private/developer_private_api_unittest.cc b/chrome/browser/extensions/api/developer_private/developer_private_api_unittest.cc
index 48d97605d5e7b..3dc95881ddb5d 100644
--- a/chrome/browser/extensions/api/developer_private/developer_private_api_unittest.cc
+++ b/chrome/browser/extensions/api/developer_private/developer_private_api_unittest.cc
@@ -72,7 +72,6 @@
 #include "extensions/browser/extension_registry.h"
 #include "extensions/browser/extension_registry_observer.h"
 #include "extensions/browser/extension_util.h"
-#include "extensions/browser/manifest_v2_experiment_manager.h"
 #include "extensions/browser/mock_external_provider.h"
 #include "extensions/browser/permissions/permissions_test_util.h"
 #include "extensions/browser/permissions/permissions_updater.h"
@@ -3415,61 +3414,6 @@ TEST_F(DeveloperPrivateApiSupervisedUserUnitTest,
     EXPECT_THAT(error, testing::HasSubstr("Child account"));
 }
 
-// Test suite for cases where the user is in the  MV2 deprecation "disabled"
-// experiment phase.
-class DeveloperPrivateApiWithMV2DeprecationDisabledUnitTest
-    : public DeveloperPrivateApiUnitTest {
- public:
-  DeveloperPrivateApiWithMV2DeprecationDisabledUnitTest() {
-    feature_list_.InitAndDisableFeature(
-        extensions_features::kExtensionManifestV2Unsupported);
-  }
-
- private:
-  base::test::ScopedFeatureList feature_list_;
-};
-
-// Extension of manifest version 2 is not supported on Android.
-#if !BUILDFLAG(IS_ANDROID)
-TEST_F(DeveloperPrivateApiWithMV2DeprecationDisabledUnitTest,
-       TestAcknowledgingAnExtension) {
-  // Add an extension that is affected by the MV2 deprecation.
-  scoped_refptr<const Extension> extension =
-      ExtensionBuilder("ext").SetManifestVersion(2).Build();
-  registrar()->AddExtension(extension.get());
-
-  ManifestV2ExperimentManager* experiment_manager =
-      ManifestV2ExperimentManager::Get(browser_context());
-  EXPECT_TRUE(experiment_manager->IsExtensionAffected(*extension));
-  EXPECT_FALSE(experiment_manager->DidUserAcknowledgeNotice(extension->id()));
-
-  base::ListValue args;
-  args.Append(extension->id());
-
-  // Call the dismiss notice function, and cancel the dismissal.
-  auto dismiss_notice_function = base::MakeRefCounted<
-      api::DeveloperPrivateDismissMv2DeprecationNoticeForExtensionFunction>();
-  dismiss_notice_function->set_source_context_type(mojom::ContextType::kWebUi);
-  dismiss_notice_function->accept_bubble_for_testing(false);
-  EXPECT_TRUE(RunFunction(dismiss_notice_function, args));
-
-  // Extension notice should NOT be marked as acknowledged.
-  EXPECT_TRUE(experiment_manager->IsExtensionAffected(*extension));
-  EXPECT_FALSE(experiment_manager->DidUserAcknowledgeNotice(extension->id()));
-
-  // Call the dismiss notice function, and accept the dismissal.
-  dismiss_notice_function = base::MakeRefCounted<
-      api::DeveloperPrivateDismissMv2DeprecationNoticeForExtensionFunction>();
-  dismiss_notice_function->set_source_context_type(mojom::ContextType::kWebUi);
-  dismiss_notice_function->accept_bubble_for_testing(true);
-  EXPECT_TRUE(RunFunction(dismiss_notice_function, args));
-
-  // Extension's notice should be marked as acknowledged.
-  EXPECT_TRUE(experiment_manager->IsExtensionAffected(*extension));
-  EXPECT_TRUE(experiment_manager->DidUserAcknowledgeNotice(extension->id()));
-}
-#endif  // !BUILDFLAG(IS_ANDROID)
-
 // Signing into transport mode and Sign outs are not supported for ChromeOS
 // hence DeveloperPrivateApiTransportModeUnitTest is not run for ChromeOS.
 #if !BUILDFLAG(IS_CHROMEOS)
diff --git a/chrome/browser/extensions/api/developer_private/developer_private_apitest.cc b/chrome/browser/extensions/api/developer_private/developer_private_apitest.cc
index 3b7c817bfcea5..ca7176f7687f3 100644
--- a/chrome/browser/extensions/api/developer_private/developer_private_apitest.cc
+++ b/chrome/browser/extensions/api/developer_private/developer_private_apitest.cc
@@ -537,126 +537,4 @@ IN_PROC_BROWSER_TEST_F(DeveloperPrivateApiTest, UninstallMultipleExtensions) {
       extension_1_id, ExtensionRegistry::EVERYTHING));
 }
 
-class DeveloperPrivateApiWithMV2DeprecationApiTest
-    : public DeveloperPrivateApiTest,
-      public testing::WithParamInterface<MV2ExperimentStage> {
- public:
-  DeveloperPrivateApiWithMV2DeprecationApiTest() {
-    experiment_stage_ = GetParam();
-
-    std::vector<base::test::FeatureRef> enabled_features;
-    std::vector<base::test::FeatureRef> disabled_features;
-    switch (experiment_stage_) {
-      case MV2ExperimentStage::kDisableWithReEnable:
-        disabled_features.push_back(
-            extensions_features::kExtensionManifestV2Unsupported);
-        break;
-      case MV2ExperimentStage::kUnsupported:
-        enabled_features.push_back(
-            extensions_features::kExtensionManifestV2Unsupported);
-    }
-
-    feature_list_.InitWithFeatures(enabled_features, disabled_features);
-  }
-
-  MV2ExperimentStage experiment_stage() const { return experiment_stage_; }
-
- private:
-  base::test::ScopedFeatureList feature_list_;
-  MV2ExperimentStage experiment_stage_;
-};
-
-// Manefiest Version 2 is not allowed in Android.
-#if !BUILDFLAG(IS_ANDROID)
-INSTANTIATE_TEST_SUITE_P(
-    ,
-    DeveloperPrivateApiWithMV2DeprecationApiTest,
-    testing::Values(MV2ExperimentStage::kDisableWithReEnable,
-                    MV2ExperimentStage::kUnsupported),
-    [](const testing::TestParamInfo<MV2ExperimentStage>& info) {
-      switch (info.param) {
-        case MV2ExperimentStage::kDisableWithReEnable:
-          return "DisableExperiment";
-        case MV2ExperimentStage::kUnsupported:
-          return "UnsupportedExperiment";
-      }
-    });
-
-// Tests that an extension's MV2 deprecation notice is marked as deprecated when
-// the function is called and by accepting the dialog, if necessary.
-// Note: we don't test cancelling the dialog since that's done extensively in
-// unit tests.
-IN_PROC_BROWSER_TEST_P(DeveloperPrivateApiWithMV2DeprecationApiTest,
-                       DismissMv2DeprecationNotice) {
-  // Load MV2 extension.
-  static constexpr char kManifest[] =
-      R"({
-           "name": "MV2 extension",
-           "manifest_version": 2,
-           "version": "0.1"
-         })";
-  TestExtensionDir test_dir;
-  test_dir.WriteManifest(kManifest);
-  const Extension* extension = LoadExtension(test_dir.UnpackedPath());
-  ASSERT_TRUE(extension);
-
-  // Verify extension is affected by the MV2 deprecation and its notice hasn't
-  // been marked as acknowledged.
-  ManifestV2ExperimentManager* experiment_manager =
-      ManifestV2ExperimentManager::Get(profile());
-  EXPECT_TRUE(experiment_manager->IsExtensionAffected(*extension));
-  EXPECT_FALSE(experiment_manager->DidUserAcknowledgeNotice(extension->id()));
-
-  // Create the dismiss notice function.
-  auto dismiss_notice_function = base::MakeRefCounted<
-      api::DeveloperPrivateDismissMv2DeprecationNoticeForExtensionFunction>();
-  std::string args = base::StringPrintf(R"(["%s"])", extension->id().c_str());
-
-  switch (experiment_stage()) {
-    case MV2ExperimentStage::kDisableWithReEnable: {
-      // The function will trigger a dialog for this stage. Add a waiter for the
-      // dialog.
-      views::NamedWidgetShownWaiter waiter(views::test::AnyWidgetTestPasskey{},
-                                           "Mv2DeprecationKeepDialog");
-      api_test_utils::SendResponseHelper response_helper(
-          dismiss_notice_function.get());
-
-      // Add a dispatcher to wait for the response since the function won't
-      // return till the dialog is accepted/canceled.
-      std::unique_ptr<ExtensionFunctionDispatcher> dispatcher(
-          new ExtensionFunctionDispatcher(profile()));
-      dismiss_notice_function->SetDispatcher(dispatcher->AsWeakPtr());
-      dismiss_notice_function->SetArgs(base::test::ParseJsonList(args));
-      dismiss_notice_function->RunWithValidation().Execute();
-
-      // Wait for the dialog and accept it.
-      auto* widget = waiter.WaitIfNeededAndGet();
-      widget->widget_delegate()->AsDialogDelegate()->AcceptDialog();
-      response_helper.WaitForResponse();
-
-      // Extension's notice should be marked as acknowledged.
-      EXPECT_TRUE(experiment_manager->IsExtensionAffected(*extension));
-      EXPECT_TRUE(
-          experiment_manager->DidUserAcknowledgeNotice(extension->id()));
-      break;
-    }
-
-    case MV2ExperimentStage::kUnsupported: {
-      std::string error = api_test_utils::RunFunctionAndReturnError(
-          dismiss_notice_function.get(), args, profile());
-      EXPECT_EQ(error, base::StringPrintf(
-                           "Cannot dismiss the MV2 deprecation notice for "
-                           "extension with ID '%s' on the unsupported stage.",
-                           extension->id().c_str()));
-
-      // Extension's notice should not be marked as acknowledged.
-      EXPECT_TRUE(experiment_manager->IsExtensionAffected(*extension));
-      EXPECT_FALSE(
-          experiment_manager->DidUserAcknowledgeNotice(extension->id()));
-      break;
-    }
-  }
-}
-#endif  // !BUILDFLAG(IS_ANDROID)
-
 }  // namespace extensions
diff --git a/chrome/browser/extensions/api/developer_private/developer_private_functions.cc b/chrome/browser/extensions/api/developer_private/developer_private_functions.cc
index 221a73b5fd8ec..45fb6f7dd468b 100644
--- a/chrome/browser/extensions/api/developer_private/developer_private_functions.cc
+++ b/chrome/browser/extensions/api/developer_private/developer_private_functions.cc
@@ -2302,9 +2302,6 @@ void DeveloperPrivateUploadExtensionToAccountFunction::OnDialogCancelled() {
 #if BUILDFLAG(IS_ANDROID)
 DEFINE_UNIMPLEMENTED_EXTENSION_FUNCTION(DeveloperPrivateLoadDirectoryFunction,
                                         "developerPrivate.loadDirectory")
-DEFINE_UNIMPLEMENTED_EXTENSION_FUNCTION(
-    DeveloperPrivateDismissMv2DeprecationNoticeForExtensionFunction,
-    "developerPrivate.dismissMv2DeprecationNoticeForExtension")
 
 ExtensionFunction::ResponseAction
 DeveloperPrivateShowSiteSettingsFunction::Run() {
@@ -2547,114 +2544,6 @@ DeveloperPrivateLoadDirectoryFunction::DeveloperPrivateLoadDirectoryFunction()
 
 DeveloperPrivateLoadDirectoryFunction::
     ~DeveloperPrivateLoadDirectoryFunction() {}
-
-DeveloperPrivateDismissMv2DeprecationNoticeForExtensionFunction::
-    DeveloperPrivateDismissMv2DeprecationNoticeForExtensionFunction() = default;
-DeveloperPrivateDismissMv2DeprecationNoticeForExtensionFunction::
-    ~DeveloperPrivateDismissMv2DeprecationNoticeForExtensionFunction() =
-        default;
-
-ExtensionFunction::ResponseAction
-DeveloperPrivateDismissMv2DeprecationNoticeForExtensionFunction::Run() {
-  std::optional<developer::DismissMv2DeprecationNoticeForExtension::Params>
-      params =
-          developer::DismissMv2DeprecationNoticeForExtension::Params::Create(
-              args());
-  EXTENSION_FUNCTION_VALIDATE(params);
-  extension_id_ = std::move(params->extension_id);
-
-  ManifestV2ExperimentManager* experiment_manager =
-      ManifestV2ExperimentManager::Get(browser_context());
-
-  // Extension must be affected by the MV2 deprecation.
-  const Extension* extension =
-      ExtensionRegistry::Get(browser_context())
-          ->GetExtensionById(extension_id_, ExtensionRegistry::EVERYTHING);
-  if (!extension) {
-    return RespondNow(Error(
-        ErrorUtils::FormatErrorMessage(kNoExtensionError, extension_id_)));
-  }
-  if (!experiment_manager->IsExtensionAffected(*extension)) {
-    return RespondNow(Error(ErrorUtils::FormatErrorMessage(
-        kExtensionNotAffectedByMV2Deprecation, extension_id_)));
-  }
-
-  MV2ExperimentStage experiment_stage =
-      experiment_manager->GetCurrentExperimentStage();
-  switch (experiment_stage) {
-    case MV2ExperimentStage::kDisableWithReEnable: {
-      // Prompt for user confirmation before dismissing the notice.
-      if (accept_bubble_for_testing_.has_value()) {
-        if (*accept_bubble_for_testing_) {
-          OnDialogAccepted();
-        } else {
-          OnDialogCancelled();
-        }
-        return AlreadyResponded();
-      }
-
-      BrowserWindowInterface* const browser =
-          ProfileBrowserCollection::GetForProfile(
-              Profile::FromBrowserContext(browser_context()))
-              ->GetLastActiveBrowser();
-      if (!browser) {
-        return RespondNow(Error(kCouldNotFindWebContentsError));
-      }
-
-      ShowMv2DeprecationKeepDialog(
-          browser, *extension,
-          base::BindOnce(
-              &DeveloperPrivateDismissMv2DeprecationNoticeForExtensionFunction::
-                  OnDialogAccepted,
-              this),
-          base::BindOnce(
-              &DeveloperPrivateDismissMv2DeprecationNoticeForExtensionFunction::
-                  OnDialogCancelled,
-              this));
-
-      return RespondLater();
-    }
-
-    case MV2ExperimentStage::kUnsupported:
-      return RespondNow(Error(ErrorUtils::FormatErrorMessage(
-          kCannotDismissExtensionOnUnsupportedStage, extension_id_)));
-  }
-}
-
-void DeveloperPrivateDismissMv2DeprecationNoticeForExtensionFunction::
-    DismissExtensionNotice() {
-  ManifestV2ExperimentManager* experiment_manager =
-      ManifestV2ExperimentManager::Get(browser_context());
-  experiment_manager->MarkNoticeAsAcknowledged(extension_id_);
-
-  // There isn't a separate observer for the MV2 acknowledged state changing,
-  // but this is the only place it's changed. Just fire the event directly.
-  DeveloperPrivateEventRouter* event_router =
-      DeveloperPrivateAPI::Get(browser_context())
-          ->developer_private_event_router();
-  if (event_router) {
-    event_router->OnExtensionConfigurationChanged(extension_id_);
-  }
-}
-
-void DeveloperPrivateDismissMv2DeprecationNoticeForExtensionFunction::
-    OnDialogAccepted() {
-  if (!browser_context()) {
-    return;
-  }
-
-  DismissExtensionNotice();
-  Respond(NoArguments());
-}
-
-void DeveloperPrivateDismissMv2DeprecationNoticeForExtensionFunction::
-    OnDialogCancelled() {
-  if (!browser_context()) {
-    return;
-  }
-
-  Respond(NoArguments());
-}
 #endif  // BUILDFLAG(IS_ANDROID)
 
 }  // namespace api
diff --git a/chrome/browser/extensions/api/developer_private/developer_private_functions.h b/chrome/browser/extensions/api/developer_private/developer_private_functions.h
index f1074021e08c5..170607b9fc028 100644
--- a/chrome/browser/extensions/api/developer_private/developer_private_functions.h
+++ b/chrome/browser/extensions/api/developer_private/developer_private_functions.h
@@ -92,9 +92,6 @@ class DeveloperPrivateAPIFunction : public ExtensionFunction {
   static constexpr char kCannotRepairNonWebstoreExtension[] =
       "Cannot repair an extension that is not installed from the Chrome Web "
       "Store.";
-  static constexpr char kCannotDismissExtensionOnUnsupportedStage[] =
-      "Cannot dismiss the MV2 deprecation notice for extension with ID '*' on "
-      "the unsupported stage.";
   static constexpr char kUserNotSignedIn[] = "User is not signed in.";
   static constexpr char kCannotUploadExtensionToAccount[] =
       "Extension with ID '*' cannot be uploaded to the user's account.";
@@ -819,13 +816,6 @@ DECLARE_UNIMPLEMENTED_EXTENSION_FUNCTION(DeveloperPrivateLoadDirectoryFunction,
                                          "developerPrivate.loadDirectory",
                                          DEVELOPERPRIVATE_LOADUNPACKEDCROS);
 
-// We don't have to port it to desktop android because MV2 is not supported
-// on Android.
-DECLARE_UNIMPLEMENTED_EXTENSION_FUNCTION(
-    DeveloperPrivateDismissMv2DeprecationNoticeForExtensionFunction,
-    "developerPrivate.dismissMv2DeprecationNoticeForExtension",
-    DEVELOPERPRIVATE_DISMISSMV2DEPRECATIONNOTICEFOREXTENSION);
-
 class DeveloperPrivateShowSiteSettingsFunction : public ExtensionFunction {
  public:
   DECLARE_EXTENSION_FUNCTION("developerPrivate.showSiteSettings",
@@ -894,46 +884,6 @@ class DeveloperPrivateLoadDirectoryFunction : public ExtensionFunction {
   // Error string if `success_` is false.
   std::string error_;
 };
-
-class DeveloperPrivateDismissMv2DeprecationNoticeForExtensionFunction
-    : public DeveloperPrivateAPIFunction {
- public:
-  DECLARE_EXTENSION_FUNCTION(
-      "developerPrivate.dismissMv2DeprecationNoticeForExtension",
-      DEVELOPERPRIVATE_DISMISSMV2DEPRECATIONNOTICEFOREXTENSION)
-  DeveloperPrivateDismissMv2DeprecationNoticeForExtensionFunction();
-
-  DeveloperPrivateDismissMv2DeprecationNoticeForExtensionFunction(
-      const DeveloperPrivateDismissMv2DeprecationNoticeForExtensionFunction&) =
-      delete;
-  DeveloperPrivateDismissMv2DeprecationNoticeForExtensionFunction& operator=(
-      const DeveloperPrivateDismissMv2DeprecationNoticeForExtensionFunction&) =
-      delete;
-
-  void accept_bubble_for_testing(bool accept_bubble) {
-    accept_bubble_for_testing_ = accept_bubble;
-  }
-
- private:
-  ~DeveloperPrivateDismissMv2DeprecationNoticeForExtensionFunction() override;
-
-  // ExtensionFunction:
-  ResponseAction Run() override;
-
-  void DismissExtensionNotice();
-
-  // Callback to run when the user accepts the keep dialog.
-  void OnDialogAccepted();
-
-  // Callback to run when the user cancels the keep dialog.
-  void OnDialogCancelled();
-
-  // The ID of the extension to be dismissed.
-  ExtensionId extension_id_;
-
-  // If true, immediately accepts the keep dialog by running the callback.
-  std::optional<bool> accept_bubble_for_testing_;
-};
 #endif  // BUILDFLAG(IS_ANDROID)
 
 }  // namespace extensions::api
diff --git a/chrome/browser/extensions/api/developer_private/extension_info_generator.cc b/chrome/browser/extensions/api/developer_private/extension_info_generator.cc
index 0d6e41d0aff4a..71c472aa201ad 100644
--- a/chrome/browser/extensions/api/developer_private/extension_info_generator.cc
+++ b/chrome/browser/extensions/api/developer_private/extension_info_generator.cc
@@ -871,8 +871,6 @@ void ExtensionInfoGenerator::FillExtensionInfo(const Extension& extension,
   CHECK(mv2_experiment_manager);
   info.is_affected_by_mv2_deprecation =
       mv2_experiment_manager->IsExtensionAffected(extension);
-  info.did_acknowledge_mv2_deprecation_notice =
-      mv2_experiment_manager->DidUserAcknowledgeNotice(extension.id());
   if (info.web_store_url.length() > 0) {
     info.recommendations_url =
         extension_urls::GetNewWebstoreItemRecommendationsUrl(extension.id())
diff --git a/chrome/browser/extensions/api/developer_private/extension_info_generator_unittest.cc b/chrome/browser/extensions/api/developer_private/extension_info_generator_unittest.cc
index fc297c3be479f..76901b179f851 100644
--- a/chrome/browser/extensions/api/developer_private/extension_info_generator_unittest.cc
+++ b/chrome/browser/extensions/api/developer_private/extension_info_generator_unittest.cc
@@ -49,8 +49,6 @@
 #include "extensions/browser/extension_prefs.h"
 #include "extensions/browser/extension_registrar.h"
 #include "extensions/browser/extension_registry.h"
-#include "extensions/browser/manifest_v2_experiment_manager.h"
-#include "extensions/browser/mv2_experiment_stage.h"
 #include "extensions/browser/permissions/permissions_test_util.h"
 #include "extensions/browser/permissions/permissions_updater.h"
 #include "extensions/browser/permissions/scripting_permissions_modifier.h"
@@ -1327,89 +1325,6 @@ TEST_F(ExtensionInfoGeneratorUnitTest, UploadAsAccountExtension_TransportMode) {
 }
 #endif  // !BUILDFLAG(IS_CHROMEOS)
 
-class ExtensionInfoGeneratorWithMV2DeprecationUnitTest
-    : public ExtensionInfoGeneratorUnitTest,
-      public testing::WithParamInterface<MV2ExperimentStage> {
- public:
-  ExtensionInfoGeneratorWithMV2DeprecationUnitTest() {
-    std::vector<base::test::FeatureRef> enabled_features;
-    std::vector<base::test::FeatureRef> disabled_features;
-
-    experiment_stage_ = GetParam();
-    switch (experiment_stage_) {
-      case MV2ExperimentStage::kDisableWithReEnable:
-        disabled_features.push_back(
-            extensions_features::kExtensionManifestV2Unsupported);
-        break;
-      case MV2ExperimentStage::kUnsupported:
-        enabled_features.push_back(
-            extensions_features::kExtensionManifestV2Unsupported);
-        break;
-    }
-
-    feature_list_.InitWithFeatures(enabled_features, disabled_features);
-  }
-  ~ExtensionInfoGeneratorWithMV2DeprecationUnitTest() override = default;
-
-  MV2ExperimentStage experiment_stage() { return experiment_stage_; }
-
- private:
-  bool ShouldUseSafetyHubFeatures() override { return false; }
-
-  base::test::ScopedFeatureList feature_list_;
-  MV2ExperimentStage experiment_stage_;
-};
-
-INSTANTIATE_TEST_SUITE_P(
-    All,
-    ExtensionInfoGeneratorWithMV2DeprecationUnitTest,
-    testing::Values(MV2ExperimentStage::kDisableWithReEnable,
-                    MV2ExperimentStage::kUnsupported),
-    [](const testing::TestParamInfo<MV2ExperimentStage>& info) {
-      switch (info.param) {
-        case MV2ExperimentStage::kDisableWithReEnable:
-          return "DisableExperiment";
-        case MV2ExperimentStage::kUnsupported:
-          return "UnsupportedExperiment";
-      }
-    });
-
-// Tests that acknowledging the MV2 deprecation notice updates the extension
-// info.
-TEST_P(ExtensionInfoGeneratorWithMV2DeprecationUnitTest,
-       DidAcknowledgeMv2DeprecationNotice) {
-  scoped_refptr<const Extension> extension =
-      ExtensionBuilder("ext").SetManifestVersion(2).Build();
-  registrar()->AddExtension(extension.get());
-
-  ManifestV2ExperimentManager* experiment_manager =
-      ManifestV2ExperimentManager::Get(browser_context());
-
-  // Extensions with manifest version 2 are affected in the other stages.
-  EXPECT_TRUE(experiment_manager->IsExtensionAffected(*extension));
-  EXPECT_FALSE(experiment_manager->DidUserAcknowledgeNotice(extension->id()));
-
-  {
-    std::unique_ptr<developer::ExtensionInfo> info =
-        GenerateExtensionInfo(extension->id());
-    EXPECT_FALSE(info->did_acknowledge_mv2_deprecation_notice);
-  }
-
-  experiment_manager->MarkNoticeAsAcknowledged(extension->id());
-
-  {
-    std::unique_ptr<developer::ExtensionInfo> info =
-        GenerateExtensionInfo(extension->id());
-    if (experiment_stage() == MV2ExperimentStage::kUnsupported) {
-      // Cannot acknowledge a notice that cannot be dismissed (unsupported
-      // stage).
-      EXPECT_FALSE(info->did_acknowledge_mv2_deprecation_notice);
-    } else {
-      EXPECT_TRUE(info->did_acknowledge_mv2_deprecation_notice);
-    }
-  }
-}
-
 #if BUILDFLAG(ENABLE_SUPERVISED_USERS)
 
 // Tests for supervised users (child accounts). Supervised users are not allowed
diff --git a/chrome/browser/resources/extensions/detail_view.ts b/chrome/browser/resources/extensions/detail_view.ts
index 74ffd5d921798..3800f47c00834 100644
--- a/chrome/browser/resources/extensions/detail_view.ts
+++ b/chrome/browser/resources/extensions/detail_view.ts
@@ -43,7 +43,6 @@ import type {ExtensionsToggleRowElement} from './toggle_row.js';
 
 class DummyDetailViewDelegate extends DummyItemDelegate {
   dismissMv2DeprecationNotice() {}
-  dismissMv2DeprecationNoticeForExtension(_id: string) {}
 }
 
 export interface ExtensionsDetailViewElement {
diff --git a/chrome/browser/resources/extensions/item_util.ts b/chrome/browser/resources/extensions/item_util.ts
index dcd5bc2a9e524..14c1a48e40b3d 100644
--- a/chrome/browser/resources/extensions/item_util.ts
+++ b/chrome/browser/resources/extensions/item_util.ts
@@ -318,7 +318,6 @@ export function createDummyExtensionInfo():
     safetyCheckWarningReason:
         chrome.developerPrivate.SafetyCheckWarningReason.UNPUBLISHED,
     isAffectedByMV2Deprecation: false,
-    didAcknowledgeMV2DeprecationNotice: false,
     canUploadAsAccountExtension: false,
   };
 }
diff --git a/chrome/browser/resources/extensions/manager.ts b/chrome/browser/resources/extensions/manager.ts
index f421f5b7c033e..ed060770dd3f6 100644
--- a/chrome/browser/resources/extensions/manager.ts
+++ b/chrome/browser/resources/extensions/manager.ts
@@ -358,8 +358,6 @@ export class ExtensionsManagerElement extends ExtensionsManagerElementBase {
         this.updateItem_(
             'extensions_', index,
             Object.assign({}, this.getData_(eventData.item_id), {
-              didAcknowledgeMV2DeprecationNotice:
-                  eventData.extensionInfo?.didAcknowledgeMV2DeprecationNotice,
               safetyCheckText: eventData.extensionInfo?.safetyCheckText,
             }));
         break;
diff --git a/chrome/browser/resources/extensions/mv2_deprecation_delegate.ts b/chrome/browser/resources/extensions/mv2_deprecation_delegate.ts
index c03b937477f52..ed84cc45ca200 100644
--- a/chrome/browser/resources/extensions/mv2_deprecation_delegate.ts
+++ b/chrome/browser/resources/extensions/mv2_deprecation_delegate.ts
@@ -4,5 +4,4 @@
 
 export interface Mv2DeprecationDelegate {
   dismissMv2DeprecationNotice(): void;
-  dismissMv2DeprecationNoticeForExtension(id: string): void;
 }
diff --git a/chrome/browser/resources/extensions/service.ts b/chrome/browser/resources/extensions/service.ts
index 0fd05277dadb9..848ddb82575bb 100644
--- a/chrome/browser/resources/extensions/service.ts
+++ b/chrome/browser/resources/extensions/service.ts
@@ -578,10 +578,6 @@ export class Service implements ServiceInterface {
         {isMv2DeprecationNoticeDismissed: true});
   }
 
-  dismissMv2DeprecationNoticeForExtension(id: string): Promise<void> {
-    return chrome.developerPrivate.dismissMv2DeprecationNoticeForExtension(id);
-  }
-
   uploadItemToAccount(id: string): Promise<boolean> {
     return chrome.developerPrivate.uploadExtensionToAccount(id);
   }
diff --git a/chrome/common/extensions/api/developer_private.webidl b/chrome/common/extensions/api/developer_private.webidl
index 486482306627c..ed195d634e4ef 100644
--- a/chrome/common/extensions/api/developer_private.webidl
+++ b/chrome/common/extensions/api/developer_private.webidl
@@ -253,7 +253,6 @@ dictionary ExtensionInfo {
   required boolean showAccessRequestsInToolbar;
   boolean pinnedToToolbar;
   required boolean isAffectedByMV2Deprecation;
-  required boolean didAcknowledgeMV2DeprecationNotice;
   required boolean canUploadAsAccountExtension;
 };
 
@@ -789,10 +788,6 @@ interface DeveloperPrivate {
   // if one is active.
   static undefined dismissSafetyHubExtensionsMenuNotification();
 
-  // Triggers the dismissal of the mv2 deprecation notice for `extensionId`.
-  static undefined dismissMv2DeprecationNoticeForExtension(
-      DOMString extensionId);
-
   // Uploads an extension to the signed in user's account and returns whether
   // the extension is actually uploaded in `callback`. If the extension is not
   // eligible for upload or if there is no signed in user, returns an error.
diff --git a/chrome/test/data/webui/extensions/detail_view_test.ts b/chrome/test/data/webui/extensions/detail_view_test.ts
index 3767a15fce5ab..a0e4c7861f35a 100644
--- a/chrome/test/data/webui/extensions/detail_view_test.ts
+++ b/chrome/test/data/webui/extensions/detail_view_test.ts
@@ -15,7 +15,6 @@ import {createExtensionInfo, MockItemDelegate, testVisible} from './test_util.js
 
 class MockDelegate extends MockItemDelegate {
   dismissMv2DeprecationNotice() {}
-  dismissMv2DeprecationNoticeForExtension(_id: string) {}
 }
 
 suite('ExtensionDetailViewTest', function() {
diff --git a/chrome/test/data/webui/extensions/test_service.ts b/chrome/test/data/webui/extensions/test_service.ts
index 8e07cf823d29e..aec25ab340619 100644
--- a/chrome/test/data/webui/extensions/test_service.ts
+++ b/chrome/test/data/webui/extensions/test_service.ts
@@ -35,7 +35,6 @@ export class TestService extends TestBrowserProxy implements ServiceInterface {
       'deleteItems',
       'dismissSafetyHubExtensionsMenuNotification',
       'dismissMv2DeprecationNotice',
-      'dismissMv2DeprecationNoticeForExtension',
       'uninstallItem',
       'downloadActivities',
       'getExtensionActivityLog',
@@ -412,10 +411,6 @@ export class TestService extends TestBrowserProxy implements ServiceInterface {
     this.methodCalled('dismissSafetyHubExtensionsMenuNotification');
   }
 
-  dismissMv2DeprecationNoticeForExtension(id: string) {
-    this.methodCalled('dismissMv2DeprecationNoticeForExtension', id);
-  }
-
   dismissMv2DeprecationNotice() {
     this.methodCalled('dismissMv2DeprecationNotice');
   }
diff --git a/extensions/browser/extension_function_histogram_value.h b/extensions/browser/extension_function_histogram_value.h
index 3f1f5cc556d60..e8e59e8e519fd 100644
--- a/extensions/browser/extension_function_histogram_value.h
+++ b/extensions/browser/extension_function_histogram_value.h
@@ -1954,7 +1954,7 @@ enum HistogramValue {
   PERMISSIONS_REMOVEHOSTACCESSREQUEST = 1891,
   ACCESSIBILITY_PRIVATE_SETCHROMEVOXFOCUS = 1892,
   DELETED_PASSWORDSPRIVATE_DELETEALLPASSWORDMANAGERDATA = 1893,
-  DEVELOPERPRIVATE_DISMISSMV2DEPRECATIONNOTICEFOREXTENSION = 1894,
+  DELETED_DEVELOPERPRIVATE_DISMISSMV2DEPRECATIONNOTICEFOREXTENSION = 1894,
   AUTOTESTPRIVATE_STARTOVERDRAWTRACKING = 1895,
   AUTOTESTPRIVATE_STOPOVERDRAWTRACKING = 1896,
   ACCESSIBILITY_PRIVATE_SCROLLATPOINT = 1897,
diff --git a/tools/metrics/histograms/metadata/extensions/enums.xml b/tools/metrics/histograms/metadata/extensions/enums.xml
index 0ec4cc1bd3b53..5c1859bf72f38 100644
--- a/tools/metrics/histograms/metadata/extensions/enums.xml
+++ b/tools/metrics/histograms/metadata/extensions/enums.xml
@@ -2816,7 +2816,7 @@ Called by update_extension_histograms.py.-->
   <int value="1893"
       label="DELETED_PASSWORDSPRIVATE_DELETEALLPASSWORDMANAGERDATA"/>
   <int value="1894"
-      label="DEVELOPERPRIVATE_DISMISSMV2DEPRECATIONNOTICEFOREXTENSION"/>
+      label="DELETED_DEVELOPERPRIVATE_DISMISSMV2DEPRECATIONNOTICEFOREXTENSION"/>
   <int value="1895" label="AUTOTESTPRIVATE_STARTOVERDRAWTRACKING"/>
   <int value="1896" label="AUTOTESTPRIVATE_STOPOVERDRAWTRACKING"/>
   <int value="1897" label="ACCESSIBILITY_PRIVATE_SCROLLATPOINT"/>
diff --git a/tools/typescript/definitions/developer_private.d.ts b/tools/typescript/definitions/developer_private.d.ts
index 2e73d1133f1f3..26e54ca663880 100644
--- a/tools/typescript/definitions/developer_private.d.ts
+++ b/tools/typescript/definitions/developer_private.d.ts
@@ -253,7 +253,6 @@ declare global {
         safetyCheckWarningReason: SafetyCheckWarningReason;
         pinnedToToolbar?: boolean;
         isAffectedByMV2Deprecation: boolean;
-        didAcknowledgeMV2DeprecationNotice: boolean;
         canUploadAsAccountExtension: boolean;
       }
 
@@ -491,8 +490,6 @@ declare global {
           site: string, updates: ExtensionSiteAccessUpdate[]): Promise<void>;
       export function dismissSafetyHubExtensionsMenuNotification(): void;
       export function dismissMv2DeprecationPanel(): void;
-      export function dismissMv2DeprecationNoticeForExtension(
-          extensionId: string): Promise<void>;
       export function uploadExtensionToAccount(extensionId: string):
           Promise<boolean>;
       export function showSiteSettings(extensionId: string): Promise<void>;
